sdk/middleware/sqecho/echo.go

```diff
@@ -3,17 +3,20 @@ package sqecho
 import (
 	"github.com/labstack/echo"
 	"github.com/sqreen/go-agent/sdk"
+	"golang.org/x/xerrors"
 )
 
 // Middleware is Sqreen's middleware function for Echo to monitor and protect
-// the requests Echo receives. It creates and stores the HTTP request record
-// both into the Echo and request contexts so that it can be later accessed from
-// handlers using `sdk.FromContext()` to perform SDK calls.
+// the requests Echo receives. In protection mode, it can block and redirect
```

Comment:
You should have 2 diferent middlewares here as you need to block IP addresses before they reach the authentication layer of the application.